### ISSUE TITLE
The generated gson type adaptor deserialize json with unknown fields

### ISSUE THREAD

**Issue by wolfjiang** [immutables/immutables#764]
There is a request that deserialize json put the unknown fields on a specified fields, how I do it ? for example, entity is Contact:

```java
@Gson.TypeAdapters
@Value.Immutable
public Contact {
    Long getId();
    String getName();
    JsonObject getOtherFields();
}
```

The json string is : 

```json
{
    "id": 1,
    "name": "my name",
    "custField1": "cust string field",
   "custField2": 2
}
```

After gson  deserialized, the expected Contact Object is(Json format):

```json
    {
        "id": 1,
       "name": "my nqme",
      "otherFields": {
           "custField1": "cust string field",
           "custField2": 2
      }
    }
```

I dig into the generated GsonAdaptorsContact.java, found that in method  `eachAttribute(JsonReader in, ImmutableDepartment.Builder builder)` , the unknown fields is skip by `in.skipValue();`. is there a configable feature than put unknows fields to a specified fields, or give me a unknown fields handler to do it myself? 
 
Sorry for my poor Englisth, any advice is welcome, thanks.

**Comment by elucash** [comment]
Thank you for raising this! I thought that something like that should be supported, but it seems that it is not implemented for Gson. For Jackson it is supported using `@JsonAnyGetter`, but not for Gson. I can think of creating a custom adapter to do this, but this would skip any advantages of using auto-generated type adapters. I'm marking this as a feature request.

**Comment by elucash** [comment]
@wolfjiang having worked with smth similar, I think it will be somewhat problematic to introduce fine-grained API to consume unknown properties, this would require handling stream, managing state and then try to apply these fields to immutable object's builder. A more tractable solution would be to discover special catch-all attributes for unknown fields and generate code in adapter to populate those. To make things clear it would look like this:

```java
@Value.Immutable interface Obj {
  int knownAttribute();
  String andThisOne();
  @Gson.Other Map<String, Object> allTheRestAttributes();
 // this could be `com.google.gson.JsonObject` or map as shown
}
```

But if a structure of mapping is completely custom/special, I would suggest to just create custom Gson `TypeAdapter` to handle it instead of introducing complex binding rules.

**Comment by wolfjiang** [comment]
@elucash cool!  @Gson.Other is a good idea to consume unknown properties, it's enough for me,  I expected it . thanks!

**Comment by wolfjiang** [comment]
Hello @elucash, glade to see that 2.6.0-alpha3 is available , but i didn't found how to handle unknown fields in alpha3, can you give me some idea? Thanks!

**Comment by elucash** [comment]
This is not implemented yet in `2.6.0-alpha3`, will be in the next milestone, currently working on completing issues for the next release, stay tuned )

**Comment by wolfjiang** [comment]
@elucash , got it. thank you very much!
